experiments/xls/stream_logging.py

```diff
@@ -21,4 +23,43 @@ def __exit__(self, exc_type, exc_value, traceback):
         xls_logger.removeHandler(self.handler)
 
     def get_log(self) -> str:
-        return self._stream.getvalue()
\ No newline at end of file
+        return self._stream.getvalue()
+
+
+class LogParser(object):
+    """
+    A parser that converts raw logs into dicts for ease of use
+    """
+
+    def __init__(self, raw_logs=[]):
+        self.raw_logs = raw_logs
+        self.logs = []
+        self.parse_logs(self.raw_logs)
+
+    def parse_logs(self, raw_logs):
+        for log in raw_logs:
+            self.parse(log)
+
+    def parse(self, raw_log):
```

Comment:
I think a docstring here would be very benefitial in understanding why these transformations happen, perhaps with an example of a raw log